src/averageAges.js

```diff
@@ -55,10 +61,29 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const mothers = people.filter(woman =>
+    people.some(mother => (woman.name === mother.mother && woman.sex === 'f')));
+
+  const children = people.filter(child =>
+    mothers.some(mother => (child.mother === mother.name)));
+
+  const onlySon = children.filter(son => son.sex === 'm');
+
+  function getAllAge(humans) {
```

Comment:
The 'getAllAge' function should be moved outside of the 'calculateAverageAgeDiff' function to improve code readability and maintainability.